.pipelines/mac-ios-spm-dev-validation-pipeline.yml

```diff
@@ -23,8 +23,8 @@
         buildType: 'specific'
         project: 'Lotus'
         definition: 995          #'definitionid' is obtained from `System.DefinitionId` of ORT CI: onnxruntime-ios-packaging-pipeline
-        buildVersionToDownload: 'latest'
-        branchName: 'main'
+        buildVersionToDownload: 'specific'
+        pipelineId: 392504
```

Comment:
using latest won't fail the pipeline task but may cause download an empty artifact if the current latest run is unsuccessful. using specific build pipeline artifacts here for now.

Reply:
Won't using a specific build also start failing as soon as that build is no longer retained? 

Should we instead add a check if that latest build was successful and fail with a meaningful error message if it was not? Not sure if devops has a direct way to test for success of the pipeline. If not adding a following step to check if the artifact was empty might be an option.

Reply:
in that case, should we also use `buildVersionToDownload: 'latestFromBranch'`?
https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/download-pipeline-artifact-v2?view=azure-pipelines#inputs

Reply:
`latest` equates to the last build of `main` doesn't it?

Reply:
the document I was referring to is a bit confusing - I just did a test build on pipeline, looks like by specifying 
```
        buildVersionToDownload: 'latest'
        branchName: 'main'
```
in the task, it will actually download the latest successful build from the pipeline specified. so it should be all good here now. will update to remove the lines here for specific build ids.

Reply:
for example: currently onnxruntime-ios-packaging-ci-pipeline latest run failed. and seems like it can download from the latest successful build from 1230: https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=394046&view=logs&j=1479c2a1-0235-5e48-6934-bc345b95bbad&t=c9c62458-fa08-538a-f94e-1f8fc22e013e&l=15